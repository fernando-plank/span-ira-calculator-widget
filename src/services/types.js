export var TAX_FILING;
(function (TAX_FILING) {
    TAX_FILING["single"] = "Single";
    TAX_FILING["joint"] = "Joint";
    TAX_FILING["hoh"] = "Head of Household";
})(TAX_FILING || (TAX_FILING = {}));
