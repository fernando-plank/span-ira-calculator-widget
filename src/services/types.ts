export type CalculatorIncomes = {
  zip: string
  owner_status: string
  household_income: string
  tax_filing: string
  household_size: string
  field?: string
  error?: string
}

export type CalculatorIncentives = {
  location: string
}

export enum TAX_FILING {
  single = 'Single',
  joint = 'Joint',
  hoh = 'Head of Household'
}

export type IncentivesResponse = {
  pos_savings: number
  tax_savings: number
  estimated_annual_savings: number
}

export type IncentivesApiResponse = {
  type: string
  program: string
  program_es: string
  item: string
  item_es?: string
  more_info_url: string
  more_info_url_es: string
  amount: number
  amount_type: string
  representative_amount?: number
  item_type: string
  owner_status: Array<string>
  ami_qualification: string
  agi_max_limit?: string | number
  filing_status?: string | number
  start_date: number
  end_date: number
  field?: string
  error?: string
  message?: string
}

export type IncentivesObject = {
  incentives: IncentivesResponse[]
}
