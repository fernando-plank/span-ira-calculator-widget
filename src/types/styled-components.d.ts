import theme from 'styles/theme'

// inferÃªncia de tipos
type Theme = typeof theme

declare module 'styled-components' {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface DefaultTheme extends Theme {
    banner: {
      mobile: string
      desktop: string
    }
    layers: {
      base: number
      menu: number
      overlay: number
      modal: number
      alwaysOnTop: number
    }
    breakpoints: {
      mobileS: string
      mobileM: string
      mobileL: string
      tablet: string
      laptop: string
      laptopL: string
      desktop: string
    }
    fonts: {
      primary: {
        family: string
        light: number
        normal: number
        bold: number
        sizes: {
          xsmall: string
          small: string
          medium: string
          large: string
          xlarge: string
          xxlarge: string
        }
      }
    }
    colors: {
      primary: string
      secondary: string
      mainBg: string
      lightBg: string
      white: string
      black: string
      lightGray: string
      gray: string
      darkGray: string
    }
    spacings: {
      xxsmall: string
      xsmall: string
      small: string
      medium: string
      large: string
      xlarge: string
      xxlarge: string
    }
    logoWhite: string
    logo: string
  }
}
