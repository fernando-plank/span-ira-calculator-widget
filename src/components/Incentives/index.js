import React from 'react';
import { TAX_FILING } from '@services/types';
import { formatCurrency } from '@utils/currency';
import * as S from './styles';
const Incentives = ({ incentives }) => (React.createElement(S.Wrapper, null,
    React.createElement("h1", null, "Incentives"),
    React.createElement("br", null),
    React.createElement("div", null,
        React.createElement("table", null,
            React.createElement("thead", null,
                React.createElement("tr", null,
                    React.createElement("th", null, "Incentive"),
                    React.createElement("th", null, "Description"),
                    React.createElement("th", null, "Tax Filing"),
                    React.createElement("th", null, "Amount"),
                    React.createElement("th", null, "Date"),
                    React.createElement("th", null, "Action"))),
            React.createElement("tbody", null, incentives.map((incentive, index) => {
                var _a;
                if (incentive.type === 'tax_credit') {
                    return (React.createElement("tr", { key: `${index}` },
                        React.createElement("td", null, incentive.item),
                        React.createElement("td", null, incentive.program),
                        React.createElement("td", null, (_a = TAX_FILING[incentive.filing_status]) !== null && _a !== void 0 ? _a : '-'),
                        React.createElement("td", null, incentive.amount_type === 'percent'
                            ? incentive.amount * 100 + '%'
                            : formatCurrency(incentive.amount)),
                        React.createElement("td", null, incentive.end_date),
                        React.createElement("td", null,
                            React.createElement("button", null,
                                React.createElement("a", { href: incentive.item_url }, "View")))));
                }
                return null;
            }))))));
export default Incentives;
