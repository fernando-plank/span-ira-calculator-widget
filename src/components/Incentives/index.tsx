import React from 'react'
import { TAX_FILING } from '@services/types'
import { formatCurrency } from '@utils/currency'

import * as S from './styles'

export type CalculatorResponseProps = {
  item: string
  program: string
  filing_status: string
  amount_type: string
  amount: number
  end_date: number
  item_url: string
  type: string
}

export type IncentivesProps = {
  incentives: CalculatorResponseProps[]
}

const Incentives = ({ incentives }: IncentivesProps) => (
  <S.Wrapper>
    <h1>Incentives</h1>
    <br />
    <div>
      <table>
        <thead>
          <tr>
            <th>Incentive</th>
            <th>Description</th>
            <th>Tax Filing</th>
            <th>Amount</th>
            <th>Date</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {incentives.map((incentive: CalculatorResponseProps, index) => {
            if (incentive.type === 'tax_credit') {
              return (
                <tr key={`${index}`}>
                  <td>{incentive.item}</td>
                  <td>{incentive.program}</td>
                  <td>{TAX_FILING[incentive.filing_status] ?? '-'}</td>
                  <td>
                    {incentive.amount_type === 'percent'
                      ? incentive.amount * 100 + '%'
                      : formatCurrency(incentive.amount)}
                  </td>
                  <td>{incentive.end_date}</td>
                  <td>
                    <button>
                      <a href={incentive.item_url}>View</a>
                    </button>
                  </td>
                </tr>
              )
            }
            return null
          })}
        </tbody>
      </table>
    </div>
  </S.Wrapper>
)

export default Incentives
