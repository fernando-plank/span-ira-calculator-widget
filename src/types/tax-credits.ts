import { ReactChild } from 'react'

export interface TaxCreditsTableData {
  item: string
  amount: number
  tooltip?: ReactChild
  lineColor?: string
}
