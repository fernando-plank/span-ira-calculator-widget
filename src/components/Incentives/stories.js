import React from 'react';
import Incentives from '.';
export default {
    title: 'Incentives',
    component: Incentives
};
const incentives = [
    {
        item: '',
        program: '',
        filing_status: '',
        amount_type: '',
        amount: 0,
        end_date: 0,
        item_url: '',
        type: ''
    }
];
export const Default = () => React.createElement(Incentives, { incentives: incentives });
