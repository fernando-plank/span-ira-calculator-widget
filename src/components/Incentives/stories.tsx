import React from 'react'
import { Story, Meta } from '@storybook/react'

import Incentives, { CalculatorResponseProps } from '.'

export default {
  title: 'Incentives',
  component: Incentives
} as Meta

const incentives: CalculatorResponseProps[] = [
  {
    item: '',
    program: '',
    filing_status: '',
    amount_type: '',
    amount: 0,
    end_date: 0,
    item_url: '',
    type: ''
  }
]

export const Default: Story = () => <Incentives incentives={incentives} />
