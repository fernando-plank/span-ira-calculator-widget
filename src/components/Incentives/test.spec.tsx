import { TAX_FILING } from '@services/types'
import { render, screen } from '@testing-library/react'

import Incentives from '.'

const calculatorResponseProps = {
  item: 'Item',
  program: 'Program',
  filing_status: 'single',
  amount_type: 'percent',
  amount: 0.3,
  end_date: 20210503,
  item_url: 'http://www.link.com',
  type: 'tax_credit'
}

describe('<Incentives />', () => {
  it('should render item', () => {
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText('Item')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render program', () => {
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText('Program')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render single status enum', () => {
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText(TAX_FILING.single)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render joint status enum', () => {
    calculatorResponseProps.filing_status = 'joint'
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText(TAX_FILING.joint)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render joint status enum', () => {
    calculatorResponseProps.filing_status = 'hoh'
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText(TAX_FILING.hoh)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render - without status', () => {
    calculatorResponseProps.filing_status = 'statusneverused'
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText('-')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render amount type percent', () => {
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText('30%')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render amount type dollar', () => {
    calculatorResponseProps.amount_type = 'currency'
    calculatorResponseProps.amount = 30
    const { container } = render(
      <Incentives incentives={[calculatorResponseProps]} />
    )

    expect(screen.getByText('$30')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
