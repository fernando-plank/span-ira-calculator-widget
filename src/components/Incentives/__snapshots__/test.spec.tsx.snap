// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Incentives /> should render - without status 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            -
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render amount type dollar 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            -
          </td>
          <td>
            $30
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render amount type percent 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            -
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render item 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            Single
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render joint status enum 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            Joint
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render joint status enum 2`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            Head of Household
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render program 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            Single
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Incentives /> should render single status enum 1`] = `
<div
  class="sc-bgqQcB cCZJli"
>
  <h1>
    Incentives
  </h1>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Incentive
          </th>
          <th>
            Description
          </th>
          <th>
            Tax Filing
          </th>
          <th>
            Amount
          </th>
          <th>
            Date
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item
          </td>
          <td>
            Program
          </td>
          <td>
            Single
          </td>
          <td>
            30%
          </td>
          <td>
            20210503
          </td>
          <td>
            <button>
              <a
                href="http://www.link.com"
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
